specification/scripts/simplify_specification.dart

```diff
@@ -0,0 +1,442 @@
+            continue;
+          }
+        }
+        continue;
+      }
+      if (gatherLine.startsCode) inCode = true;
+      if (inCode) {
+        if (gatherLine.endsCode) inCode = false;
+        continue;
+      }
+      final trimmedGatherLine = gatherLine.trimLeft();
+      if (trimmedGatherLine.startsList) {
+        // We do not gather a nested itemized list into the current item.
+        // Finalize the current item.
+        this[itemIndex] = buffer.toString();
+        // Set up the first item of the nested itemized list.
+        itemIndex = _findItem(gatherIndex + 1);
+        itemLine = this[itemIndex];
+        buffer = StringBuffer(itemLine!);
+        gatherIndex = itemIndex;
+        continue;
+      }
+      if (gatherLine.endsList) {
+        // At the end of the outermost itemized list: Done.
+        this[itemIndex] = buffer.toString();
+        return gatherIndex;
+      }
+      final foundItem = trimmedGatherLine.isItem;
+      final foundEnd = trimmedGatherLine.endsList;
+      if (foundItem || foundEnd) {
+        // Current `\item` has ended, transfer the data.
+        this[itemIndex] = buffer.toString();
+        if (foundItem) {
+          // Another `\item` coming, set up.
+          buffer = StringBuffer(gatherLine);
+          itemIndex = gatherIndex;
+          itemLine = this[itemIndex]; // Restore the `itemLine` invariant.
+          continue;
+        } else {
+          // `foundEnd` is true.
+          // Gather lines after the nested itemized list, if any. Note
+          /// that `itemLine` does not contain `\item`, but it's treated
+          /// as if it did contain `\item`.
+          itemIndex = _findText(gatherIndex + 1);
+          itemLine = this[itemIndex]; // Restore the `itemLine` invariant.
+          if (itemLine!.endsList) {
+            // No extra lines, at the end of the outermost itemized list: Done.
+            return itemIndex + 1;
+          } else if (itemLine.trimLeft().isItem) {
+            // No extra lines, starting a new item.
+            buffer = StringBuffer(itemLine);
+            gatherIndex = itemIndex;
+            continue;
+          } else {
+            // Some extra text found, gather it.
+            buffer = StringBuffer(itemLine); // Not starting with `\item`.
+            gatherIndex = itemIndex;
+            continue;
+          }
+        }
+      }
+      // `gatherLine` is text belonging to the current `\item`.
+      final spacing = insertSpace ? ' ' : '';
+      final endsInPercent = gatherLine.endsWith('%');
+      final String addLine;
+      if (endsInPercent) {
+        addLine = gatherLine.substring(0, gatherLine.length - 1).trimLeft();
+        insertSpace = false;
+      } else {
+        addLine = gatherLine.trimLeft();
+        insertSpace = true;
+      }
+      if (addLine.isNotEmpty) buffer.write('$spacing$addLine');
+      this[gatherIndex] = null;
+    }
+    throw "_gatherItems reached end of text";
```

Comment:
Nit: I would prefer `StateError("_gatherItems reached end of text")` or something similar over the raw string as the parameter to `throw`. It might make error handling slightly easier in the future.

Reply:
Done.